src/highdicom/sr/templates.py

```diff
@@ -1955,19 +3248,454 @@ def __init__(
+            Graphic type of image region
+        referenced_sop_instance_uid: str, optional
+            SOP Instance UID of the referenced instance, which may be a
+            segmentation image, source image for the region or segmentation, or
+            RT struct, depending on `reference_type`
+        referenced_sop_class_uid: str, optional
+            SOP Class UID of the referenced instance, which may be a
+            segmentation image, source image for the region or segmentation, or
+            RT struct, depending on `reference_type`
+
+        Returns
+        -------
+        List[highdicom.sr.PlanarROIMeasurementsAndQualitativeEvaluations]
+            Sequence of content items for each matched measurement group
+
+        """  # noqa: E501
+        # FIXME: reference type and referenced sop instance uid
+        measurement_group_items = self._find_measurement_groups()
+        sequences = []
+        for group_item in measurement_group_items:
+            if group_item.template_id is not None:
+                if group_item.template_id != '1410':
+                    continue
+            else:
+                contains_rois = _contains_planar_rois(group_item)
+                if not(contains_rois):
+                    continue
+
+            matches = []
+            if finding_type is not None:
+                matches_finding = _contains_code_items(
+                    group_item,
+                    name=codes.DCM.Finding,
+                    value=finding_type
+                )
+                matches.append(matches_finding)
+            if finding_site is not None:
+                matches_finding_sites = _contains_code_items(
+                    group_item,
+                    name=codes.SCT.FindingSite,
+                    value=finding_site
+                )
+                matches.append(matches_finding_sites)
+            if tracking_uid is not None:
+                matches_tracking_uid = _contains_text_items(
+                    group_item,
+                    name=codes.DCM.TrackingUniqueIdentifier,
+                    value=tracking_uid
+                )
+                matches.append(matches_tracking_uid)
+            if graphic_type is not None:
+                if isinstance(graphic_type, GraphicTypeValues):
+                    matches_graphic_type = _contains_scoord_items(
+                        group_item,
+                        name=codes.DCM.ImageRegion,
+                        graphic_type=graphic_type
+                    )
+                else:
+                    matches_graphic_type = _contains_scoord3d_items(
+                        group_item,
+                        name=codes.DCM.ImageRegion,
+                        graphic_type=graphic_type
+                    )
+                matches.append(matches_graphic_type)
+
+            seq = PlanarROIMeasurementsAndQualitativeEvaluations.from_sequence(
+                [group_item]
+            )
```

Comment:
Didn't we decide to eagerly perform type conversion? In which case this is unnecessary and `group_item` can be returned directly?

Reply:
The `group_item` is just an individual `ContentItem`, but the `PlanarROIMeasurementsAndQualitativeEvaluations` is a `Template`, which is a `ContentSequence` containing one or more `ContentItem` objects.

Reply:
Ah right...